pysc2/pysc2/env/remote_sc2_env.js

```diff
@@ -1 +1,226 @@
+      visualize: Whether to pop up a window showing the camera and feature
+          layers. This won't work without access to a window manager.
+      step_mul: How many game steps per agent step (action/observation). None
+          means use the map default.
+      realtime: Whether to use realtime mode. In this mode the game simulation
+          automatically advances (at 22.4 gameloops per second) rather than
+          being stepped manually. The number of game loops advanced with each
+          call to step() won't necessarily match the step_mul specified. The
+          environment will attempt to honour step_mul, returning observations
+          with that spacing as closely as possible. Game loops will be skipped
+          if they cannot be retrieved and processed quickly enough.
+      replay_dir: Directory to save a replay.
+      replay_prefix: An optional prefix to use when saving replays.
+
+    Raises:
+      ValueError: if the race is invalid.
+      ValueError: if the resolutions aren't specified correctly.
+      ValueError: if lan_port is a sequence but its length != 4.
+      */
+    super()
+
+    if (_only_use_kwargs) {
+      throw new ValueError('Alll arguments must be passed as keyword arguments.')
+    }
+
+    if (agent_interface_format == null) {
+      throw new ValueError('Please specify agent_interface_format.')
+    }
+
+    if (!race) {
+      race = sc2_env.Race.random
+    }
+
+    const map_inst = map_name && maps.get(map_name)
+    this._map_name = map_name
+    this._game_info = null
+    this._num_agents = 1
+    this._discount = discount
+    if (map_inst) {
+      this._step_mul = step_mul || map_inst.step_mul
+    } else {
+      this._step_mul = step_mul || 8
+    }
+    this._realtime = realtime
+    this._last_step_time = null
+    if (replay_dir) {
+      this._save_replay_episodes = 1
+    } else {
+      this._save_replay_episodes = 0
+    }
+    this._replay_dir = replay_dir
+    this._replay_prefix = replay_prefix
+
+    this._score_index = -1 // Win/loss only.
+    this._score_multiplier = 1
+    this._episode_length = sc2_env.MAX_STEP_COUNT
+    this._ensure_available_actions = false
+    this._discount_zero_after_timeout = false
+
+    this._run_config = run_configs.get()
+    this._parallel = run_parallel.RunParallel() // Needed for multiplayer.
+    this._in_game = false
+    this._action_delay_fns = [null]
+
+    const required_raw = visualize
+    const interface = this._get_interface(agent_interface_format, required_raw)
+    // need to be fixed from python  
+    // if isinstance(lan_port, all_collections_generated_classes.Sequence):
+    if (isinstance(lan_port, )) {
+      if (lan_port.lenth != 4) {
+        throw new ValueError('lan_port sequence must be of length 4')
+      }
+      const ports = lan_port[:]
+    } else {
+      let ports =[]
+      for (let i = 0; i < 4; i++) {
+        ports.push(lan_port + i) // 2 * num players *in the game*.
+      }
+    }
```

Comment:
let ports
  if (Array.isArray(lan_port)) {
    if (lan_port.lenth != 4) {
      throw new ValueError('lan_port sequence must be of length 4')
    }
    ports = lan_port
  } else {
    ports =[]
    for (let p = 0; p < 4; p++) {
      ports.push(lan_port + p) // 2 * num players *in the game*.
    }
  }